File: leetcode/1431_kids_with_the_greatest_number_of_candies/solution.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max_value = *max_element(candies.begin(), candies.end());
        vector<bool> result;
        result.reserve(candies.size());

        for (int num : candies) {
            result.push_back(num + extraCandies >= max_value);
        }

        return result;
    }
};

File: leetcode/1431_kids_with_the_greatest_number_of_candies/solution.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max_value = max(candies)
        result = []

        for num in candies:
            result.append(num + extraCandies >= max_value)

        return result